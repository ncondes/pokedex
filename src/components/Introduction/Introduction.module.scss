@use '@/scss/mixins' as *;

.introduction {
   margin-bottom: 1rem;
   text-align: center;
   font-size: 1rem;
   font-weight: 300;
}

.introduction_extended {
   display: none;

   @include breakpoint-up('md') {
      display: block;
   }
}
