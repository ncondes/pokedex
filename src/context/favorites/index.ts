export { FavoritesContext } from './FavoritesContext'
export { FavoritesProvider } from './FavoritesProvider'
