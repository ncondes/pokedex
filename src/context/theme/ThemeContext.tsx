'use client'

import { createContext } from 'react'

interface Props {
   theme: 'light' | 'dark'
   toggleTheme: () => void
   useTheme: (className: string) => string
}

export const ThemeContext = createContext({} as Props)
