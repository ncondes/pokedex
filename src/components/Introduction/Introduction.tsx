'use client'

import { usePathname } from 'next/navigation'
import { ThemeContext } from '@/context/theme'
import { useContext } from 'react'
import styles from './Introduction.module.scss'

export const Introduction = () => {
   const pathname = usePathname()
   const { useTheme } = useContext(ThemeContext)
   const fontColorClassName = useTheme('font')

   return (
      <>
         {pathname === '/' ? (
            <p className={`${styles.introduction} ${fontColorClassName}`}>
               Discover and explore a vast collection of diverse and fascinating
               Pokemon creatures.{' '}
               <span className={styles.introduction_extended}>
                  Easily search for your favorite Pokemon, learn about their
                  unique abilities, and uncover their captivating stories. Start
                  your adventure now and delve into the magical world of
                  Pokemon!
               </span>
            </p>
         ) : (
            <p className={`${styles.introduction} ${fontColorClassName}`}>
               Welcome to your Favorite Pokemon Section! Discover, admire, and
               celebrate the Pokemon that hold a special place in your heart.
            </p>
         )}
      </>
   )
}
