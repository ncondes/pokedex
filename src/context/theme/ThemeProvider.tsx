'use client'

import { FC, PropsWithChildren, useState } from 'react'
import { ThemeContext } from './ThemeContext'
import { getLocalTheme, setLocalTheme } from '@/utils'

const INITIAL_THEME = getLocalTheme() as 'light' | 'dark'

export const ThemeProvider: FC<PropsWithChildren> = ({ children }) => {
   const [theme, setTheme] = useState<'light' | 'dark'>(INITIAL_THEME)

   const toggleTheme = () => {
      const t = theme === 'light' ? 'dark' : 'light'
      setTheme(t)
      setLocalTheme(t)
   }

   const useTheme = (className: string): string => `${theme}-${className}`

   return (
      <ThemeContext.Provider value={{ theme, toggleTheme, useTheme }}>
         {children}
      </ThemeContext.Provider>
   )
}
