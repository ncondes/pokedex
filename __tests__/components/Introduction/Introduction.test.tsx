import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { usePathname } from 'next/navigation'
import { ThemeProvider } from '@/context/theme'
import { Introduction } from '@/components/Introduction'

jest.mock('next/navigation', () => ({
   usePathname: jest.fn(),
}))

describe('Introduction', () => {
   it('renders the introduction text when the pathname is root', () => {
      ;(usePathname as jest.Mock).mockReturnValue('/')

      const { getByText } = render(
         <ThemeProvider>
            <Introduction />
         </ThemeProvider>
      )

      expect(
         getByText(
            'Discover and explore a vast collection of diverse and fascinating Pokemon creatures.'
         )
      ).toBeInTheDocument()
   })

   it('renders the introduction text when the pathname is not root', () => {
      ;(usePathname as jest.Mock).mockReturnValue('/favorites')

      const { getByText } = render(
         <ThemeProvider>
            <Introduction />
         </ThemeProvider>
      )

      expect(
         getByText(
            'Welcome to your Favorite Pokemon Section! Discover, admire, and celebrate the Pokemon that hold a special place in your heart.'
         )
      ).toBeInTheDocument()
   })

   it('should match the snapshot', () => {
      ;(usePathname as jest.Mock).mockReturnValue('/')

      const { container } = render(
         <ThemeProvider>
            <Introduction />
         </ThemeProvider>
      )

      expect(container).toMatchSnapshot()
   })
})
