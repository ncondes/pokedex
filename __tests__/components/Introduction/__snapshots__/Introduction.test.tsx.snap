// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction should match the snapshot 1`] = `
<div>
  <p
    class="introduction light-font"
  >
    Discover and explore a vast collection of diverse and fascinating Pokemon creatures.
     
    <span
      class="introduction_extended"
    >
      Easily search for your favorite Pokemon, learn about their unique abilities, and uncover their captivating stories. Start your adventure now and delve into the magical world of Pokemon!
    </span>
  </p>
</div>
`;
